karate-core/src/main/java/com/intuit/karate/core/ScenarioOutline.java

```diff
@@ -101,13 +104,13 @@ public List<Scenario> getScenarios(FeatureRuntime fr) {
             if (selectedForExecution) {
                 Table table = examples.getTable();
                 if (table.isDynamic()) {
-                    Scenario scenario = toScenario(table.getDynamicExpression(), -1, table.getLineNumberForRow(0), examples.getTags());
+                    Scenario scenario = toScenario(table.getDynamicExpression(), examples.getIndex(), -1, table.getLineNumberForRow(0), examples.getTags());
```

Comment:
so following up from my previous comment - suggest since in this loop we know how many examples and example tables are there - we will be able to emit an extra boolean if this is the last scenario selected for execution within an outline

if that solves your current ask - I would just do that, as I really don't see a need for tracking counts of examples and tables :|

Reply:
Sounds good I can definitely make that simplification. What about the `scenarioResults` though? What do you feel would be the best way to get that information from an API perspective.

Reply:
For now I just removed the table index code in [6e84b5d](https://github.com/karatelabs/karate/pull/2636/commits/6e84b5dba4486382422273dc5fdacaec4548aa2f). Didn't want to push too far ahead before we finalized something.